ipywidgets/widgets/widget.py

```diff
@@ -134,7 +134,11 @@ def _call_widget_constructed(widget):
     @staticmethod
     def handle_comm_opened(comm, msg):
         """Static method, called when a widget is constructed."""
-        widget_class = import_item(str(msg['content']['data']['widget_class']))
+        class_name = str(msg['content']['data']['widget_class'])
+        if class_name in Widget.widget_types:
+            widget_class = Widget.widget_types[class_name]
+        else:
+            widget_class = import_item(class_name)
```

Comment:
If you do this in this PR, I guess you could also change the widget controller to instantiate buttons and axis using the registry.